crates/shared/src/state.rs

```diff
@@ -159,6 +159,13 @@ where
             <Types::BlockPayload as BlockPayload<Types>>::from_bytes(encoded_txns, metadata);
         let txn_commitments = block_payload.transaction_commitments(metadata);
 
+        let last_nonempty_view = if txn_commitments.is_empty() {
+            self.parent_block_references.last_nonempty_view
+        } else {
+            // This block is non-empty
+            Some(quorum_proposal.view_number)
```

Comment:
IIRC in the previous commit you updated `self.parent_block_references.last_nonempty_view` as well. Should we keep it?

Reply:
Sorry, not sure I understand what you mean

Reply:
As discussed today this actually looks good!